dotnet/src/SemanticKernel.Skills/Skills.MsGraph/EmailSkill.cs

```diff
@@ -79,4 +108,35 @@ public async Task SendEmailAsync(string content, SKContext context)
         string[] recipientList = recipients.Split(new[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries);
         await this._connector.SendEmailAsync(subject, content, recipientList);
     }
+
+    /// <summary>
+    /// Get email messages as the body
+    /// </summary>
+    [SKFunction("Get email messages with specified optional clauses used to query messages in Microsoft Graph")]
+    [SKFunctionContextParameter(Name = Parameters.Select, Description = "The select parameter used to select the properties to retrieve in the response.")]
+    [SKFunctionContextParameter(Name = Parameters.Filter, Description = "The filter parameter used to filter the results based on specific criteria.")]
+    [SKFunctionContextParameter(Name = Parameters.Top, Description = "The top parameter used to limit the number of results returned in the response.")]
+    [SKFunctionContextParameter(Name = Parameters.Skip, Description = "The skip parameter used to skip a certain number of results in the response.")]
+    [SKFunctionContextParameter(Name = Parameters.OrderBy, Description = "The orderby parameter used to sort the results based on specific properties in either ascending or descending order.")]
+    public async Task<string> GetEmailMessagesAsync(SKContext context)
+    {
+        this._logger.LogInformation("Getting email messages with query options select: '{0}', filter: '{1}', top: '{2}', skip:'{3}', orderby: '{4}'",
+            context.Variables.Get(Parameters.Select, out string select),
+            context.Variables.Get(Parameters.Filter, out string filter),
+            context.Variables.Get(Parameters.Top, out string top),
+            context.Variables.Get(Parameters.Skip, out string skip),
+            context.Variables.Get(Parameters.OrderBy, out string orderBy)
+        );
+
+        IUserMessagesCollectionPage emailsPage = await this._connector.GetMessagesAsync(
+            topClause: string.IsNullOrWhiteSpace(top) ? null : int.Parse(top),
+            skipClause: string.IsNullOrWhiteSpace(skip) ? null : int.Parse(skip),
+            filterClause: filter,
+            orderByClause: orderBy,
+            selectClause: select
+        );
+
+        // Serialize all email content either by their body or subject returned by the page and return as a string as skill output
```

Comment:
I would like to return an object collection (Email) from the skill but the only examples I have seen are all string's returned by the Skill.  I'm not sure if that is by design the the contract for Skills can ultimately only return string to represent raw text. 

For example I see the OrgHierarchy connector: https://github.com/microsoft/semantic-kernel/blob/ed9f38f364ad04e27a11491e2eaa3829de4560d7/dotnet/src/SemanticKernel.Skills/Skills.MsGraph/Connectors/OrganizationHierarchyConnector.cs#L40 return a IEnumerable<string> but no corresponding Skill for that function: https://github.com/microsoft/semantic-kernel/blob/ed9f38f364ad04e27a11491e2eaa3829de4560d7/dotnet/src/SemanticKernel.Skills/Skills.MsGraph/OrganizationHierarchySkill.cs#L13

 
 
Would be great to get some clarity here.

@craigomatic for your input, please feel free to tag others.